src/xattr.c

```diff
@@ -527,7 +527,7 @@ static int scoutfs_xattr_set(struct dentry *dentry, const char *name,
 		return -EPERM;
 
 	bytes = sizeof(struct scoutfs_xattr) + name_len + size;
-	xat = kmalloc(bytes, GFP_NOFS);
+	xat = kzalloc(bytes, GFP_NOFS);
```

Comment:
Again this can be huge.  No need to zero the whole buffer when most of it is overwritten.  Just initialize the padding in the if (value) block which prepares the xat header to be stored in items.